product_variant_configurator/models/product_configurator.py

```diff
@@ -0,0 +1,275 @@
+# Copyright 2015 Oihane Crucelaegui - AvanzOSC
+# Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2016 ACSONE SA/NV
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
+
+import logging
+from odoo import _, api, exceptions, fields, models
+from odoo.addons import decimal_precision as dp
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductConfigurator(models.AbstractModel):
+    _name = 'product.configurator'
+    _description = 'Product Configurator'
+
+    product_tmpl_id = fields.Many2one(
+        string='Product Template',
+        comodel_name='product.template',
+        auto_join=True)
+    product_attribute_ids = fields.One2many(
+        comodel_name='product.configurator.attribute',
+        domain=lambda self: [("owner_model", "=", self._name)],
+        inverse_name='owner_id', string='Product attributes', copy=True)
+    price_extra = fields.Float(
+        compute='_compute_price_extra',
+        digits=dp.get_precision('Product Price'),
+        help="Price Extra: Extra price for the variant with the currently "
+             "selected attributes values on sale price. eg. 200 price extra, "
+             "1000 + 200 = 1200.")
+    product_id = fields.Many2one(
```

Comment:
This is generating an issue with inventory adjustments:
```
(...)
File "/home/lois/odoo/12.0/odoo/addons/stock/models/stock_inventory.py", line 280, in _get_inventory_lines_values
    GROUP BY product_id, location_id, lot_id, package_id, partner_id """ % domain, args)
(...)
psycopg2.ProgrammingError: column reference "product_id" is ambiguous
LINE 1: SELECT product_id, sum(quantity) as product_qty, location_id...
```
To reproduce just install stock and try to do a inventory adjustment

Reply:
It's reported since a lot...

https://github.com/odoo/odoo/pull/34387

We will have to do fully qualified SQL expression in OCB. In master, it has been replaced by a proper ORM `read_group`.